lib/board.rb

```diff
@@ -0,0 +1,40 @@
+class Board
+  def initialize
+    @board = [1, 2, 3, 4, 5, 6, 7, 8, 9] 
+  end
+
+  # def play
+  #   puts draw_board
+  # end
+
+  def draw_board
+    "\n" +
+    " #{@board[0]} | #{@board[1]} | #{@board[2]} " + "\n" +
+    "===+===+===" + "\n" +
+    " #{@board[3]} | #{@board[4]} | #{@board[5]} " + "\n" +
+    "===+===+===" + "\n" +
+    " #{@board[6]} | #{@board[7]} | #{@board[8]} " + "\n" +
+    "\n"
+  end
+
+  def free_spot?(spot)
+    @board[spot - 1] != 'X' && @board[spot - 1] != 'O'
+  end
+
+  def valid_marker?(player_mark)
+    player_mark == 'X' || player_mark == 'O'
+  end
+
+  def mark_board(player_mark, spot)
+    if !valid_marker?(player_mark)      
+      puts "\n I don't recognize #{player_mark}. Try again with X or O.\n"
```

Comment:
I've been thinking about this comment and while I want to keep the error handling methods (`valid_marker?` and `free_spot?`), I agree with you and I'm not sure the Board class is ultimately where I want to do something about these errors. 

My thought process: if the spot is taken, I want the user to have the opportunity to try again. Since I'm not housing any of the input or output functionality in the board class, I currently can't accept another response. My thought is that I remove everything except `@board[spot - 1] = player_mark` from the `mark_board` method, and handle this behavior using the truthy methods in another class. Thoughts?

Reply:
you already have a function to check if a "spot is free" so you can check for that before attempting to "mark a spot". So, you should be able to validate any input prior to calling this method. The specifics around validation could change if board is ever re-used elsewhere (doubtful for TTT, but you never know).  IF that's the case, why not just have `mark_board` throw an error if anyone attempts to mark a spot that is not free?